### DISCUSSION TITLE
Joint variable expansion generation

### DISCUSSION THREAD

**Discussion by jwward96** [jonathf/chaospy#437] [General]
Hi! I've been working to make some parallels to this toolbox in MATLAB to aid in some work I'm doing with stochastic NMPC. I understand the implemented methods of generating the expansions for individual variables but have a question on what happens when generating a joint expansion.

I read in the code that for independent variables you can generate the expansion for each RV and use a tensor product to get the joint expansion. So for two gaussian RV's with a 3rd order expansion you get 4 terms for each RV. The tensor product would then return a 16 term expansion but with Chaospy I noticed it only returns a 10 term joint expansion. 

I'm certain there is just some method of simplifying redundant terms or possibly terms that are not linearly independent. Would someone be able to explain what simplifications are made or point towards a reference for me to read more about this?

**Comment by jonathf** [comment]
https://chaospy.readthedocs.io/en/master/api/chaospy.generate_expansion.html
See `cross_truncation`.

Set the value to `np.inf` to get the full tensor product.

**Reply by jwward96** [reply]
That’s exactly what I was looking for! It seems I went too far into the code right away. Guess it pays to just read the top level function call!
